substrate/frame/transaction-storage/src/lib.rs

```diff
@@ -98,10 +109,13 @@ pub mod pallet {
 			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
 			+ GetDispatchInfo
 			+ From<frame_system::Call<Self>>;
-		/// The currency trait.
-		type Currency: ReservableCurrency<Self::AccountId>;
+		/// The currency type for this pallet.
+		type Currency: FnMutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
```

Comment:
see discussion https://github.com/paritytech/polkadot-sdk/issues/226#issuecomment-1691692395

```suggestion
		/// The fungible type for this pallet.
		type Fungible: FnMutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
```